src/platform/ESP32/NetworkCommissioningDriver.h

```diff
@@ -136,6 +136,35 @@ class ESPWiFiDriver final : public WiFiDriver
     uint16_t mLastDisconnectedReason;
 };
 
+class ESPEthernetDriver final : public EthernetDriver
+{
+public:
+    class EthernetNetworkIterator final : public NetworkIterator
+    {
+    public:
+        EthernetNetworkIterator(ESPEthernetDriver * aDriver) : mDriver(aDriver) {}
+        size_t Count() { return 1; }
+        bool Next(Network & item) { return true; }
```

Comment:
This doesn't make sense.  This is returning `true` but not initializing `item, so the caller will get random data...

Reply:
Assuming there is only one network then can we return false here?

Reply:
No, if there's only one network you need to return `true` exactly once, and that one time you need to fill in the fields of `item` with information about that network: the `networkID`, `networkIDLen`, and "connected" boolean.

Reply:
I have added this change